service/frontend/workflowHandler.go

```diff
@@ -1754,6 +1767,12 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		token.PersistenceToken = nil
 	}
 
+	// calculate event store version based on if branch token exist
+	token.EventStoreVersion = persistence.EventStoreVersionV2
```

Comment:
can you move this logic into `queryHistory` ?

Reply:
done